### PR TITLE
Miscminionaccessories

### PR DESCRIPTION
**Pull request by SkippZzMods**
Adds a couple of summoner accessories, Haunted Amethyst & Stone of the Drowned.

https://trello.com/c/KjiFKBl5/396-haunted-amethyst-stone-of-the-drowned-05

nothing I can think of

### CODE REVIEW

File: Content/Items/Dungeon/Accessories.StoneOfTheDrowned.cs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	public class StoneOfTheDrowned : SmartAccessory
+	{
+		internal float slotsUsed; // used so you only summon the amount of slots just gained. Otherwise would populate all of the players remaining slots
+		// i love private terraria methods which require reflection to access
+		public static MethodInfo? playerItemCheckShoot_Info;
+		public static Action<Player, int, Item, int>? playerItemCheckShoot;
+
+		public override string Texture => AssetDirectory.JungleItem + Name;
+
+		public StoneOfTheDrowned() : base("Stone of the Drowned", "+30 barrier\nIncreases your number of max minions by 2 when you have no barrier\nAutomatically summons two slots worth of minions if there is a valid weapon in your inventory") { }
```
Comment (ScalarVector1) [inline Content/Items/Dungeon/Accessories.StoneOfTheDrowned.cs:23]: lets shorten this to something like:

Increases your max number of minions by 2 while you have no barrier
Automatically re-summons 2 minions when you reach 0 barrier

File: Content/Items/Dungeon/Accessories.StoneOfTheDrowned.cs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+		public override void Load()
+		{
+			playerItemCheckShoot_Info = typeof(Player).GetMethod("ItemCheck_Shoot", BindingFlags.NonPublic | BindingFlags.Instance);
+			//Here we cache this method for performance
+			playerItemCheckShoot = (Action<Player, int, Item, int>)Delegate.CreateDelegate(
+				typeof(Action<Player, int, Item, int>), playerItemCheckShoot_Info);
+
```
Comment (ScalarVector1) [inline Content/Items/Dungeon/Accessories.StoneOfTheDrowned.cs:31]: extraneous linebreak

File: Content/Items/Dungeon/Accessories.StoneOfTheDrowned.cs [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+		public override void SafeSetDefaults()
+		{
+			Item.value = Item.sellPrice(gold: 2);
+			Item.rare = ItemRarityID.Orange;
+		}
+
+		public override void SafeUpdateEquip(Player Player)
+		{
+			Player.GetModPlayer<BarrierPlayer>().maxBarrier += 30;
+			if (Player.GetModPlayer<BarrierPlayer>().barrier <= 0)
```
Comment (ScalarVector1) [inline Content/Items/Dungeon/Accessories.StoneOfTheDrowned.cs:43]: should probably leave a linebreak before this, these are seperate effects

File: Content/Items/Dungeon/Accessories.StoneOfTheDrowned.cs [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+			{
+				Player.maxMinions += 2;
+
+				if (Player.slotsMinions < Player.maxMinions && slotsUsed < 2)
```
Comment (ScalarVector1) [inline Content/Items/Dungeon/Accessories.StoneOfTheDrowned.cs:47]: Should we seperate out a helper method for finding a valid summon item?

File: Content/Items/Jungle/Accessories.HauntedAmethyst.cs [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+		public override void SafeUpdateEquip(Player Player)
+		{
+			if (Player.statLife < Player.statLifeMax2 * 0.5f)
+			{
+				Player.maxMinions += 1;
+
+				if (Player.slotsMinions < Player.maxMinions && slotsUsed < 1)
+				{
+					Item summoningItem = null;
```
Comment (ScalarVector1) [inline Content/Items/Jungle/Accessories.HauntedAmethyst.cs:46]: The answer is yes, yes we should, since its boilerplated here :)

**Review by ScalarVector1 — CHANGES_REQUESTED** [review]

File: Content/Items/Dungeon/Accessories.StoneOfTheDrowned.cs (hunk 4 above)
Comment (SkippZzMods) [reply Content/Items/Dungeon/Accessories.StoneOfTheDrowned.cs:47]: no its specific for the item

Comment (SkippZzMods) [reply Content/Items/Dungeon/Accessories.StoneOfTheDrowned.cs:47]: or maybe

Comment (SkippZzMods) [reply Content/Items/Dungeon/Accessories.StoneOfTheDrowned.cs:47]: the spawning of the minions would also need to be a part of the helper method, which i dont think is ideal and very niche

**Review by ScalarVector1 — APPROVED** [review]
